File: Algorithms/InsertionSort.cpp
#include "InsertionSort.h"
InsertionSort::InsertionSort() {

}
void InsertionSort::Sort(int* arr,int count) {
	for (int i = 1; i < count;i++) {
		int compareNum = arr[i];
		int prevNumIndex = i - 1;
		while (prevNumIndex>-1 && arr[prevNumIndex] > compareNum) {
			int next = arr[prevNumIndex + 1];
			int cur = arr[prevNumIndex];
			arr[prevNumIndex + 1] = arr[prevNumIndex];
			prevNumIndex--;
		}
		arr[prevNumIndex + 1] = compareNum;
	}
}

File: src/sort/insertionSort.ts
import {Sort} from "./Sort";
export default class InsertionSort extends Sort{
    sort(array: number[]): void | number[] {
        for(let i = 1; i <array.length;i++){
            let compareNum = array[i];
            let j = i-1;
            while(j > -1 && array[j] > compareNum){
                array[j+1] = array[j];
                j--;
            }
            array[j+1] = compareNum;
        }
    }
}